internal/supervisor/installation.go

```diff
@@ -156,10 +167,16 @@ func NewInstallationSupervisorSchedulingOptions(balanceInstallations bool, clust
 // Shutdown performs graceful shutdown tasks for the installation supervisor.
 func (s *InstallationSupervisor) Shutdown() {
 	s.logger.Debug("Shutting down installation supervisor")
+	s.cache.stop <- true
```

Comment:
It is probably an unlikely edge case but if I am not mistaken this will block forever if we stop the supervisor before `Do` was ever called, as nothing will be listening to the channel.

We could start the cache goroutine in `NewInstallationSupervisor`. I am guessing you wanted to avoid that so that we do not hammer cache initialization on every restart, tho this may cause a tricky bug that will result in a non-graceful shutdown.

I think alternatively we can close the channel and this should trigger the receive.

Reply:
Good catch thanks. I took a slightly different approach that I think should resolve all possible issues here. Let me know what you think.